File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return None
        d=deque([root])

        while d:
            node=d.popleft()
            node.left,node.right=node.right,node.left
            if node.left:
                d.append(node.left)
            if node.right:
                d.append(node.right)

        return root
        

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> in;
    void inorder(TreeNode* node){
        if(!node){
            return;
        }
        inorder(node->left);
        in.push_back(node->val);
        inorder(node->right);

    }
    int kthSmallest(TreeNode* root, int k) {
        inorder(root);
        return in[k-1];
        
    }
};
